core/src/save/entity.rs

```diff
@@ -0,0 +1,80 @@
+use crate::Position;
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+#[serde(tag = "id")]
+pub enum EntityData {
+    #[serde(rename = "minecraft:item")]
+    Item(ItemEntityData),
+    #[serde(rename = "minecraft:arrow")]
+    Arrow(ArrowEntityData),
+
+    /// Fallback type for unknown entities
+    #[serde(other)]
+    Unknown,
+}
+
+/// Common entity tags.
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct BaseEntityData {
+    #[serde(rename = "Pos")]
+    pub position: Vec<f64>,
+    #[serde(rename = "Rotation")]
+    pub rotation: Vec<f32>,
+    #[serde(rename = "Motion")]
+    pub velocity: Vec<f64>,
+}
+
+impl BaseEntityData {
+    /// Reads the position and rotation fields. If the fields are invalid, None is returned.
+    pub fn read_position(self: &BaseEntityData) -> Option<Position> {
+        if self.position.len() == 3 && self.rotation.len() == 2 {
+            Some(Position {
+                x: self.position[0],
+                y: self.position[1],
+                z: self.position[2],
+                yaw: self.rotation[0],
+                pitch: self.rotation[1],
+                on_ground: true,
+            })
+        } else {
+            None
+        }
+    }
+}
+
+/// Represents a single item, without slot information.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ItemData {
+    #[serde(rename = "Count")]
+    pub count: u8,
+    #[serde(rename = "id")]
+    pub item: String,
+}
+
+/// Data for an Item entity (`minecraft:item`).
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct ItemEntityData {
+    // Inherit base entity data
+    #[serde(flatten)]
+    pub entity: BaseEntityData,
+
+    // Item-specific tags
+    #[serde(rename = "Age")]
+    pub age: i16,
+    #[serde(rename = "PickupDelay")]
+    pub pickup_delay: u8,
+    #[serde(rename = "Item")]
+    pub item: ItemData,
+}
+
+/// Data for an Arrow entity (`minecraft:arrow`).
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct ArrowEntityData {
+    // Inherit base entity data
+    #[serde(flatten)]
+    pub entity: BaseEntityData,
+
+    // Arrow-specific tags
+    #[serde(rename = "crit")]
+    pub critical: u8,
```

Comment:
For some reason, it's not able to deserialize `TAG_Byte` as `bool` in those structs. It logs an error saying `Region file contains invalid NBT: invalid type: integer 0, expected a boolean`. I'm not sure why since `LevelData` contains multiple `bool` fields that load just fine.

Reply:
Odd, maybe the value changed to a TAG_Byte in 1.13? I know the map format documentation on both the Minecraft wiki and wiki.vg is outdated. I'll inspect the NBT when I get home.

Reply:
Hmm... well, it's probably a problem with hematite-nbt. We could file an issue there, but I'm not sure of what the cause could be...

Reply:
I identified a reproducible test case and submitted an issue there: https://github.com/PistonDevelopers/hematite_nbt/issues/43

Reply:
Great. Would you like to merge now or after that issue is resolved?

Reply:
I'm going to add some more tests and documentation to this PR first, but I don't think we should block on that issue. I'll add a todo to change that crit field to bool.